x/evm/keeper/state_transition.go

```diff
@@ -357,6 +357,14 @@ func (k *Keeper) ApplyMessageWithConfig(ctx sdk.Context, msg core.Message, trace
 	stateDB := statedb.New(ctx, k, txConfig)
 	evm := k.NewEVM(ctx, msg, cfg, tracer, stateDB)
 
+	leftoverGas := msg.Gas()
+	if evm.Config.Debug {
```

Comment:
can you add a comment on what is happening here?

Reply:
done